analyses/fusion-summary/01-fusion-summary.Rmd

```diff
@@ -203,9 +232,133 @@ allFuseEwing %>%
+  select(Sample, FusionName) %>%
+  distinct()
+
+three_prime_kinase_outframe
+```
+
+Let's look at these just to be sure the results are as expected.
+```{r}
+# `BS_KE56MMY0 ARHGEF2--NTRK1` one does and will be captured in the `three_prime_kinase_inframe` list, but `BS_B1C6GZ84 CHIC2--PDGFRA2` does not retain the kinase domain, so we do not want to add it.
+intersect(three_prime_kinase_outframe[,c("Sample", "FusionName")], three_prime_kinase_inframe[,c("Sample", "FusionName")])
+```
+
+```{r}
+# Which fusions are not in-frame?
+three_prime_kinase_outframe <- allFuseLGAT %>%
+  filter(grepl("Kinase", Gene1B_anno) & Fusion_Type != "in-frame") %>%
+  distinct()
+# Check they don't retain the kinase domain. They do not, so leave out.
+table(three_prime_kinase_outframe$DomainRetainedGene1B)
+```
+
+Next, filter all fusions for 5' kinase fusions which have reciprocal fusions and retain the kinase domain.
+Keep these for the final output file.
+```{r}
+# Keep those with kinase domain retained and fusion in-frame - keep this list
+five_prime_domain_intact <- allFuseLGAT %>%
+  filter(grepl("Kinase", Gene1A_anno) & DomainRetainedGene1A == "Yes" & Fusion_Type == "in-frame") %>%
+  select(Sample, FusionName) %>%
+  distinct()
+```
+
+Next, filter all fusions for 5' kinase fusions which have lost the kinase domain for reciprocal fusions which have a kinase domain intact and are in-frame.
+Keep these for the final output file.
+```{r}
+# First, get fusions with 5' kinases which lost the kinase domain and have a reciprocal, then add the reciprocal fusion.
+five_prime_domain_lost <- allFuseLGAT %>%
+  filter(grepl("Kinase", Gene1A_anno) & reciprocal_exists == "TRUE") %>%
+  select(Sample, FusionName, Gene1A, Gene1B, Fusion_Type, DomainRetainedGene1A) %>%
+  filter(DomainRetainedGene1A == "No") %>%
+  distinct() %>%
+  mutate(reciprocal = paste(Gene1B, Gene1A, sep ="--"))
+
+# Then, select the in-frame reciprocals which have the kinase domain in tact.
+five_prime_reciprocals <- five_prime_domain_lost %>% 
+  select(Sample, FusionName = reciprocal) %>%
+  left_join(allFuseLGAT, by = c("Sample", "FusionName")) %>%
+  filter(Fusion_Type == "in-frame" & DomainRetainedGene1B == "Yes")
+
+# Add back the reciprocal of the reciprocal (the 5' kinase fusion) and keep this list
+five_prime_kinase_keep <- five_prime_reciprocals %>%
+  mutate(reciprocal = paste(Gene1B, Gene1A, sep ="--")) %>%
+  select(Sample, reciprocal) %>%
+  rename(FusionName = reciprocal) %>%
+  distinct()
+```
+
+Rbind lists for final table of LGAT fusions of interest
+```{r}
+# Rbind lists for final table of LGAT fusions of interest
+subsetFuseLGAT <- bind_rows(nonkinaseLGAT,
+                            three_prime_kinase_inframe[,c("Sample", "FusionName")], 
+                            five_prime_domain_intact[,c("Sample", "FusionName")], 
+                            five_prime_kinase_keep[,c("Sample", "FusionName")]) %>%
+  select(Sample, FusionName) %>%
```

Comment:
It looks like you've already selected all the data frames to `c("Sample", "FusionName")` so is this `select()` here is a bit redundant except for `nonkinaseLGAT`. Since bind_rows go by column names, you could keep this `select()` here and delete all the instances of `[,c("Sample", "FusionName")]` above and it will also be fine but shorter to read.

Reply:
ahh, cool. got it!